Realm/ObjectStore/object_store.cpp

```diff
@@ -355,6 +365,23 @@ void ObjectStore::create_tables(Group *group, Schema &target_schema, bool update
             set_primary_key_for_object(group, target_object_schema->name, "");
         }
     }
+    return to_delete;
+}
+
+void ObjectStore::remove_properties(Group *group, Schema &target_schema, std::unordered_map<std::string, Property> to_delete) {
+    for (auto& target_object_schema : target_schema) {
+        TableRef table = table_for_object_type(group, target_object_schema.name);
+        ObjectSchema current_schema(group, target_object_schema.name);
+        size_t deleted = 0;
+        for (auto& current_prop : current_schema.properties) {
+            current_prop.table_column -= deleted;
+            if (to_delete.find(current_prop.name) != to_delete.end()) {
```

Comment:
This doesn't appear to work if multiple classes have properties with the same name.